### DISCUSSION TITLE
Recent compilation error

### DISCUSSION THREAD

**Discussion by vaisoliviu** [codefori/vscode-ibmi#2395] [Q&A] [answered]
Hello,

I am experiencing a problem when I try to compile programs and files:

Could this be related to the message I have here?

![image](https://github.com/user-attachments/assets/56a412cd-b5c0-4ca9-b8a9-4f49bf77fe68)

Also if I click on this to force enable it, i get the no external description found for all the files in my program.

Any suggestion would be greatly appreciated.

<hr />

⚠️ **REMOVE THIS LINE AND ANY SENSITIVE INFORMATION BELOW!** ⚠️

|Context|Version|
|-|-|
|Code for IBM i version|2.14.4|
|Visual Studio Code version|1.95.3|
|Operating System|win32_x64|

<details>
<summary>Active extensions</summary>

```
Code for IBM i Walkthroughs (vscode-ibmi-walkthroughs): 0.5.0
Configuration Editing (configuration-editing): 1.0.0
Db2 for IBM i (vscode-db2i): 1.6.3
Emmet (emmet): 1.0.0
Git (git): 1.0.0
Git Base (git-base): 1.0.0
GitHub (github): 0.0.1
GitHub Authentication (github-authentication): 0.0.2
IBM i Debug (ibmidebug): 2.0.1
IBM i Notebooks (vscode-ibmi-notebooks): 0.0.6
IBM i Project Explorer (vscode-ibmi-projectexplorer): 2.12.0
JSON Language Features (json-language-features): 1.0.0
Merge Conflict (merge-conflict): 1.0.0
Node Debug Auto-attach (debug-auto-launch): 1.0.0
RPGLE (vscode-rpgle): 0.26.15
TODO Highlight (vscode-todo-highlight): 1.0.5
TypeScript and JavaScript Language Features (typescript-language-features): 1.0.0
```
</details>

<hr />

<details>
<summary>Remote system</summary>

|Setting|Value|
|-|-|
|IBM i OS|V7R4M0|
|Tech Refresh|10|
|CCSID Origin|65535|
|Runtime CCSID|297|
|Default CCSID|297|
|SSHD CCSID|?|
|cqsh|true|
|SQL|Enabled
|Source dates|Disabled

### Enabled features

|/QOpenSys/pkgs/bin|/usr/bin|/QSYS.lib/ILEDITOR.lib|/QSYS.LIB|/QIBM/ProdData/IBMiDebugService/bin|/QOpenSys/QIBM/ProdData/JavaVM/jdk80|
|-|-|-|-|-|-|
|bash|attr|GETNEWLIBL.PGM|QZDFMDB2.PGM|startDebugService.sh|64bit|
|chsh|iconv|||||
|grep|setccsid|||||
|ls|tar|||||
|md5sum|uname|||||
|sort||||||
|stat||||||
</details>

<details>
<summary>Shell env</summary>

```bash
BUILDLIB=COMP
CURLIB=COMP
HOME=/home/infova
HOST=VSR29422
LIBLS= *all
LOGIN=infova
LOGNAME=infova
MAIL=/var/spool/mail/infova
OLDPWD=/home/INFOVA
PATH=/QOpenSys/usr/bin:/usr/ccs/bin:/QOpenSys/usr/bin/X11:/usr/sbin:.:/usr/bin
PWD=/home/infova
SHELL=/QOpenSys/pkgs/bin/bash
SHLVL=1
SSH_CLIENT= 
SSH_CONNECTION= 
TZ=<CET>-1<CEST>,M3.5.0,M10.5.0
USER=infova
USERNAME=INFOVA
WORKDIR=/home/infova
_=/QOpenSys/usr/bin/env
```
</details>

<details>
<summary>Variants</summary>

```json
{
  "american": "#@$",
  "local": "£à$"
}
```
</details>

**Comment by worksofliam** [comment]
@vaisoliviu I converted this to a discussion. This happens when the IBM Project Explorer extension is activated. I suggest you disable that extension, reload VS Code, and then the typical library list will be shown.

**Reply by vaisoliviu** [reply]
Thanks allot @worksofliam. Still have an issue though:
![image](https://github.com/user-attachments/assets/ebc0dbe8-54d0-4739-ba0d-061dc787654c)

This is the error from the terminal:
bash: line 1: /home/infova/.vscode/cqsh_1: No such file or directory

**Reply by worksofliam** [reply]
@vaisoliviu In your connection settings, under the Source tab, change your default deployment method to 'No Default' and go from there.

<img width="803" alt="image" src="https://github.com/user-attachments/assets/03061c60-5ec9-4e19-9670-7f3c04f0d429">

**Reply by vaisoliviu** [reply, marked as answer]
@worksofliam Thanks, the issue was actually caused by an action that I modified some time ago, it is weird that it stopped working now.
I have an env variable that i use when i compile to insert the current date as text into the objects description. That was giving the errors.

**Reply by vaisoliviu** [reply]
As for the default Deployment Method, although I choose the 'No default' option, as soon as I exit the screen, it goes to *blanks and i have the "Default deployment method is set but not usable in your environment." each time I compile.

**Reply by SanjulaGanepola** [reply]
@vaisoliviu Looks like you are running into the same issue mentioned [here](https://github.com/codefori/vscode-ibmi/issues/2398). I have a pending PR to address this.
